src/lambda/wizards/samDeployWizard.ts

```diff
@@ -357,23 +388,82 @@ export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
                     'AWS.samcli.deploy.s3Bucket.prompt',
                     'Enter the AWS S3 bucket to which your code should be deployed'
                 ),
+                value: initialValue || '',
+                matchOnDetail: true,
                 ignoreFocusOut: true,
-                prompt: localize(
-                    'AWS.samcli.deploy.s3Bucket.region',
-                    'S3 bucket must be in selected region: {0}',
+            },
+            // preloaded "busy screen" message
+            // using a const in case we want to make this clickable as a way to abort the lookup.
+            items: [
+                {
+                    label: loadingBuckets
+                }
+            ],
+        })
+
+        // throw up a busy screen while the S3 buckets are loading
+        picker.makeQuickPickBusy(quickPick)
+        quickPick.show()
+
+        let listBucketsResponse: S3.ListBucketsOutput | undefined
```

Comment:
Please try to parcel some of this portion out to a separate method. It isn't easy to follow.